cupy/core/elementwise.pxi

```diff
@@ -365,12 +365,13 @@ cdef list _get_out_args(list out_args, tuple out_types, tuple out_shape,
 
 
 cdef list _get_out_args_with_params(
-        list out_args, tuple out_types, tuple out_shape, tuple out_params):
+        list out_args, tuple out_types, tuple out_shape, tuple out_params, size):
     cdef ParameterInfo p
     if not out_args:
         for p in out_params:
             if p.raw:
-                raise ValueError('Output array size is Undecided')
+                if size is not None:
```

Comment:
Sorry, this conditional expression I added was wrong (and opposite). Correct one is "if size is None:". will update including your comment.